src/abstractdataframe/join.jl

```diff
@@ -337,10 +352,27 @@ function Base.join(df1::AbstractDataFrame,
                 push!(leftonly_ixs, l_ix)
             end
         end
-        return joiner.dfl[leftonly_ixs, :]
+        joined = joiner.dfl[leftonly_ixs, :]
     else
         throw(ArgumentError("Unknown kind of join requested: $kind"))
     end
+
+    if !(indicator == nothing)
```

Comment:
use of `indicator_cols` can cause problems when `_left` or `_right` name was used in the "other" table. Then with `makeunique=true` they would get renamed and you would:

1) remove the wrong column below
2) create `left` and `right` using a wrong column

A quick fix would be to use column number not column name.

Reply:
Actually you have to create column names in a better way if I am not mistaken anyway. So this code can stay, but column creation above should be fixed. The reason is that with `makeunique=false` if there is a duplicate column `_left` or `_right` in the other `DataFrame` join will fail as it will detect a duplicate.

(if I have missed something and you handle this correctly then please let me know, it would be best if an appropriate test covering all corner cases with offending column names were created)

Reply:
I think I'm ok with this actually -- since when I `hcat` in `_left` and `_right`, I have `makeunique=true`, there's no risk of name collisions. Then on line 296 I pull the resulting column names out so if a conflict and rename occurred, I'd get the updated column name (`_left_x` or whatever). It's those pulled names I used in the later delete. 

But the obvious answer to check when I fill out the tests, which I was just holding off till we agreed on syntax. :) Will work on those shortly.

Reply:
OH! now I see, thanks for explaining. Will use column numbers in next commit.

Reply:
Even with column numbers you still need careful column names as temporary names could clash on join with normal names in the other table (and cause an error if `makeunique=false` or cause an unnecessary rename of columns if `makeunique=true`).

The simplest that comes to my mind is to create a `Set` of existing column names in both data frames and check if `_left` is in it. If yes add `X` at the end to get `_leftX` and so on till you get a unique name. The reason I recommend to add `X` is that we know that when DataFrames.jl tries to make unique column names it adds `_{number}` suffix which will not clash with `X` character (so you can use whatever you like as long as it is not a number 😄). The same with `_right`.

Reply:
@bkamins Oh, I thought by "column numbers" you meant positional column numbers. Those are pretty safe, no? In the final result, can't I count on the first helper column (currently `_left`) being in position `ncol(df1) + 1` (where `df1` is initial input dataframe -- just `ncol(df1)` after I add the column), and the second (`_right`) in `ncol(joined) - 2`?

Reply:
Now it looks good. @nalimilan - can you please have a look?

Reply:
Looks good now, but the only way to be sure is to have tests. AFAICT they currently cover this corner case, right?